server/market/market.go

```diff
@@ -1040,17 +1050,77 @@ func (m *Market) processOrder(rec *orderRecord, epoch *EpochQueue, notifyChan ch
 	// ID, this also prevents orders with the same ID.
 	// TODO: Prevent commitment reuse in general, without expensive DB queries.
 	ord := rec.order
+	oid := ord.ID()
+	user := ord.User()
 	commit := ord.Commitment()
 	m.epochMtx.RLock()
 	otherOid, found := m.epochCommitments[commit]
 	m.epochMtx.RUnlock()
 	if found {
 		log.Debugf("Received order %v with commitment %x also used in previous order %v!",
-			ord, commit, otherOid)
+			oid, commit, otherOid)
 		errChan <- ErrInvalidCommitment
 		return nil
 	}
 
+	// Whether an order is a taker depends on type, and for limit orders it
+	// depends on force and rates.
+	bestBuy, midGap, bestSell := m.rates()
+	likelyTaker := func(ord order.Order) bool {
+		lo, ok := ord.(*order.LimitOrder)
+		if !ok || lo.Force == order.ImmediateTiF {
+			return true
+		}
+		// Infer from best buy/sell to be conservative, ignoring other side.
+		switch {
+		case lo.Sell:
+			return lo.Rate < bestSell
+		default:
+			return lo.Rate > bestBuy
+		}
```

Comment:
Any trades in the spread are considered takers and limited. Super conservative. I'm not sure how a trader would feel. 20 hrs is a long time though.

Reply:
Correct.  It was a tough call with this "guess", but I wanted to play it safe.  BTW, this switch is why an empty market says an order is likely taker.  Could be changed with

```go
		switch {
		case midGap == 0:
			return false
		case lo.Sell:
			return lo.Rate < bestSell
		default:
			return lo.Rate > bestBuy
		}
```

And I had it like that for a bit, but decided that with an empty market, you actually are fairly likely to be taker if anyone else submits a matching order into the epoch.  I do see the issue that is creating with the trade_simnet_test.go tests.  Thinking about solutions...

Reply:
I think the max of 1200 lots is a bit repressive. What's a whale to do?

Reply:
One more thing while I have your year. The quantities are a bit hard for human eyes with all the zeros. Maybe lots is fine?

Reply:
Switched it so the order rate must actually *cross the spread* into the other side of the book.  This is way less conservative.